x/exchange/msg_test.go

```diff
@@ -0,0 +1,2237 @@
+package exchange
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/testutil/assertions"
+)
+
+const (
+	emptyAddrErr = "empty address string is not allowed"
+	bech32Err    = "decoding bech32 failed: "
+)
+
+func TestAllMsgsGetSigners(t *testing.T) {
+	// getTypeName gets just the type name of the provided thing, e.g. "MsgGovCreateMarketRequest".
+	getTypeName := func(thing interface{}) string {
```

Comment:
Can rename this to getMsgTypeName to make it a bit clearer. Can we make the type of thing stricter? I think it's only interacting with sdk.Msg.

Reply:
It's an anonymous function used only for the purposes of this unit test. One benefit of them is it's easier to keep them shorter since they only need to be know in the context of this one unit test. In here, we are only concerned about the message type names. And there's no real reason to make it stricter since the extra limitations don't buy us anything .

Reply:
The benefit is clarity, and it makes it easier to understand as you trace through the test.

Reply:
Also, as it's written, it's not specific to messages. It's available to use on anything. E.g. It then can be used on things that haven't yet implemented `sdk.Msg`, but will need to at some point.